version.go

```diff
@@ -38,8 +44,8 @@ func init() {
 	buildInfo, ok := debug.ReadBuildInfo()
```

Comment:
Since there was some confusion in the PR it would be good to document the point of this code especially as it relates to the replace directive

Reply:
This should be addressed below... let me know if you were thinking of something else @jameshartig !